##  Natal origin estimation of foraging green turtle using genome-wide SNPs:

This github repository was made to post the codes that were used in the study Hamabata et al. (in review).

***

- **Raw read data** All the read files (fastq format) were deposit in GenBank BioProject accession
- **Reference genome** Reference squence (fasta format) of green turtle, <i>Chelonia mydas</i> was downloaded from [here](https://www.ncbi.nlm.nih.gov/assembly/GCF_015237465.2#/st) (rCheMyd1.pri.v2).

***

1. [Read quality fitering](https://github.com/tmkhmbt/foraging_turtle_group_assignment/blob/main/read_quality_filtering.md)
2. [Mapping and creat catalogue of SNPs](https://github.com/tmkhmbt/foraging_turtle_group_assignment/blob/main/mapping_and_bam_creat.md)
3. [Population structure](https://github.com/tmkhmbt/foraging_turtle_group_assignment/blob/main/Population%20structure.md)
4. [Extract SNPs for group assignment]()

Although, in this Github repository, we add the link to jump the sofware information, we used the Biocontainers Singularity Images that are available in [NIG Supercomputer system](https://sc.ddbj.nig.ac.jp/en/) for some software without installing in our own environment. 
